init/pipeline.py

```diff
@@ -1,43 +1,55 @@
 import dlt
-from dlt.sources.helpers import requests
 
+from dlt.sources.helpers.rest_client import paginate
+from dlt.sources.helpers.rest_client.auth import BearerTokenAuth
+from dlt.sources.helpers.rest_client.paginators import HeaderLinkPaginator
 
-@dlt.source
-def source(api_secret_key=dlt.secrets.value):
-    return resource(api_secret_key)
-
-
-def _create_auth_headers(api_secret_key):
-    """Constructs Bearer type authorization header which is the most common authorization method"""
-    headers = {"Authorization": f"Bearer {api_secret_key}"}
-    return headers
+# This pipeline demonstrates how to use the dlt REST client for extracting data from the GitHub API.
```

Comment:
think of those templates as generic. they are not examples. it is OK if we use Github just move this comment into the dlt.resource